Sources/MapboxMaps/Ornaments/OrnamentsManager.swift

```diff
@@ -39,271 +18,165 @@ public enum OrnamentVisibility: String, Equatable {
 
-    internal func ornament(at position: OrnamentPosition) -> [Ornament] {
-        return ornaments.filter({ $0.position == position })
-    }
+        let scaleBarViewConstraints = constraints(with: scalebarView, position: options.scaleBarPosition, margins: options.scaleBarMargins)
+        constraints.append(contentsOf: scaleBarViewConstraints)
 
-    internal func ornament(withType type: OrnamentType) -> [Ornament] {
-        return ornaments.filter({ $0.type == type })
-    }
+        let attributionButtonConstraints = constraints(with: attributionButton, position: options.attributionButtonPosition, margins: options.attributionButtonMargins)
+        constraints.append(contentsOf: attributionButtonConstraints)
 
-    private func addToView(ornaments: [Ornament]) {
-        ornaments.forEach {
-            addToView(ornament: $0)
-        }
-    }
+        // Activate new constraints
+        NSLayoutConstraint.activate(constraints)
 
-    private func removeFromView(ornaments: [Ornament]) {
-        ornaments.forEach {
-            removeFromView(ornament: $0)
-        }
+        logoView.isHidden = !options._logoViewIsVisible
+        scalebarView.isHidden = options.scaleBarVisibility == .hidden
+        compassView.isHidden = options.compassVisibility == .hidden
+        attributionButton.isHidden = !options._attributionButtonIsVisible
     }
 
-    // swiftlint:disable:next cyclomatic_complexity
-    private func addToView(ornament: Ornament) {
-        if ornament.view == nil {
-            ornament.view = ornament.type.makeOrnament(for: view, visibility: ornament.visibility)
-        }
-        guard let ornamentView = ornament.view else {
-            assert(false, "Couldn't create view for an ornament")
-            return
-        }
-        var constraints = [NSLayoutConstraint]()
-        view.addSubview(ornamentView)
+    func constraints(with view: UIView, position: OrnamentPosition, margins: CGPoint) -> [NSLayoutConstraint] {
+        let universalLayoutGuide = layoutGuide(for: view)
 
-        switch ornament.position {
+        switch position {
         case .topLeft:
-            constraints.append(contentsOf: [
-                ornamentView.leadingAnchor.constraint(equalTo: universalLayoutGuide.leadingAnchor,
-                                                      constant: ornament.margins.x),
-                ornamentView.topAnchor.constraint(equalTo: universalLayoutGuide.topAnchor,
-                                                  constant: ornament.margins.y)
-            ])
+            return [
+                view.leftAnchor.constraint(equalTo: universalLayoutGuide.leftAnchor,
+                                          constant: margins.x),
+                view.topAnchor.constraint(equalTo: universalLayoutGuide.topAnchor,
+                                              constant: margins.y)
+                ]
+
         case .topCenter:
-            print("top center")
+            print("topCenter")
```

Comment:
why are we just printing here? What is the purpose of the empty case?

Reply:
It's just not implemented. (and wasn't prior to this change as well). We should discuss whether we should implement or remove the enum case.

Reply:
I feel like since this PR is now a refactor, we should really consider removing the unused enum cases. Or at the minimum, consolidate the cases and put a `print("Not supported at this time")`